### ISSUE TITLE
M10 Migration issues

### ISSUE THREAD

**Issue by jhpedemonte** [maqetta/maqetta#3820]
Migration issues to investigate/handle for M10 release:
- [x] Are old users migrated correctly?  For both regular login and LDAP?
  - Check that user properties (particularly "login" and "email") are set correctly.
  - Check that users can access their files from Orion Navigator.
- [x] Are pre-M10 users' file rights correct? Can they view their projects/files in Orion Navigator? See f7eda0823d46dae8ceadf2e5eda8a966696146e1.
- [x] Are custom widgets created in M9 accessible/usable in M10?
- [x] Deploy updated **siteConfig** files.
- [x] Disable Orion indexer (see [instructions](https://github.com/maqetta/maqetta/issues/3820#issuecomment-16942494)).

@JonFerraiolo @billreed63 @peller

**Comment by JonFerraiolo** [comment]
Regarding custom widgets in M9, I just added a new `customWidgetSpec` property to all of the `*_widgets.json` files for custom widgets. Right now, if that property is not present, then M10 will print a console warning message and simply ignore that custom widget. Therefore, what we need to test is create a custom widget in M9 and make sure that the console message appears in M10 and that the widget doesn't appear in widget palette. Note that if any content exists that uses an old custom widget, well, sorry.

(UPDATED) I tried things out (in Eclipse) where I took a custom widget created using M9 on Maqetta.org and installed it in my M10 workspace, and put an HTML file that used this widget in my M10 workspace. The page does not parse, so this message appears in small letters across the top of the canvas: 

Unable to parse HTML source. See console for error. Please switch to "Display Source" mode and correct the error.

And then the console first shows a message saying the older version of the custom widget isn't supported, and shows an error message from dojo parser about not being able to parse the custom widget.

@peller: One problem is that the source editor is hosed and I couldn't remove the offending line of HTML there. I had to go to Orion Navigator to remove the line, and then I could continue editing the file.

**Comment by jhpedemonte** [comment]
Can we do not any bit of migration in the code, to add `customWidgetSpec` if it's not there?  Or is it not worth it?

**Comment by jhpedemonte** [comment]
I created an Eclipse workspace in M9, then created a custom widget composed of HTML elements and Dijit TabContainer (surrounded by DIV).  I created a new desktop app and dropped custom widget in there, no problem.

Using same workspace with latest code failed to find the custom widget (not in widget palette).  Loading of file with the custom widget failed as well.

**Comment by jhpedemonte** [comment]
Still running into issues with the Orion indexer when running with a workspace that contains a large number of users.  Specifically, the indexer locks on EclipsePreference objects while doing its thing.  Later, Maqetta requests come in that under the covers attempt to get a lock on the same EclipsePreferences -- request is held up until the indexer momentarily relinquishes control, causing many requests to be delayed for many seconds each.

Granted, there may be an issue with what we're doing during each request on the Maqetta side, but it's bad for us to have the Orion indexer blocking us like this.  Solution is to, once again, disable the indexer.

---

Last time I tried this, I couldn't just simply remove the *.search jar (issues with dependencies?).  So the hack I came up with was to remove certain files from within the jar; when the indexer tries to setup, it cannot copy the files and therefore doesn't start the indexer.
1. Delete the _solr_ config files:
   
   ```
   zip -d plugins/org.eclipse.orion.server.search_* solrconf/\*
   ```
2. Delete the _solr metadata_ from the users workspace (dir may not yet exist if this is a new workspace dir):
   
   ```
   rm -r users/.metadata/.plugins/org.eclipse.orion.server.core.search
   ```

**Comment by jhpedemonte** [comment]
For the record:  while testing migration of M9 workspaces, several of us noticed an issue where we lost all of our M9 projects and were left with only the default "project1" (and default files).  The culprit, as above, was with the Orion indexer.

Since we're moving from Orion 0.5 (in M9) to 2.0 (in M10), Orion does a migration of **Users.prefs** in the **.metadata** dir.  However, the indexer is running at the same time (against a large number of user accounts);  since it puts locks on the metadata files, the migration runs _extremely slow_.

While [debugging the pert issues](https://github.com/maqetta/maqetta/issues/3820#issuecomment-16942494), we killed the server and restarted it.  This had the effect of stopping the migration before it was done, and many M9 user accounts were erased as a result from **Users.prefs**.  Then when we logged in, Maqetta found no files for our accounts and simply created the default project.

This is all fixes as above, by disabling the Orion indexer.
